File: fixtures/sample.go
// Fixture: exercise go highlighting + symbol picker.

package main

import (
	"errors"
	"fmt"
	"strings"
)

const (
	MaxRetries = 3
	Greeting   = "hello, world"
)

type Status int

const (
	Online Status = iota
	Offline
	Away
)

type Greeter interface {
	Greet() string
}

type Person struct {
	Name string
	Age  int
}

func (p Person) Greet() string {
	return fmt.Sprintf("hi, %s! you are %d", p.Name, p.Age)
}

func double(x int) int {
	return x * 2
}

func shout(s string) (string, error) {
	if s == "" {
		return "", errors.New("empty input")
	}
	return strings.ToUpper(s) + "!!!", nil
}

func main() {
	p := Person{Name: "Ada", Age: 36}
	for i := 0; i < MaxRetries; i++ {
		if i%2 == 0 {
			fmt.Println(Greeting, "->", p.Greet())
		} else if msg, err := shout(Greeting); err == nil {
			fmt.Println(msg)
		}
	}
	_ = double(21)
}

File: fixtures/sample.py
"""Fixture: exercise python highlighting + symbol picker."""

from dataclasses import dataclass
from typing import Iterable

MAX_RETRIES = 3
GREETING = "hello, world"


@dataclass
class Person:
    name: str
    age: int

    def greet(self) -> str:
        return f"hi, {self.name}! you are {self.age}"


class Greeter:
    def __init__(self, prefix: str = "yo"):
        self.prefix = prefix

    def greet_all(self, people: Iterable[Person]) -> list[str]:
        return [f"{self.prefix} {p.greet()}" for p in people]


def double(x: int) -> int:
    return x * 2


def main() -> None:
    p = Person(name="Ada", age=36)
    g = Greeter()
    for i in range(MAX_RETRIES):
        if i % 2 == 0:
            print(GREETING, "->", p.greet())
        else:
            print(GREETING.upper() + "!!!")
    print(g.greet_all([p, Person("Linus", 54)]))
    assert double(21) == 42


if __name__ == "__main__":
    main()
